server/msgtrace.go

```diff
@@ -496,49 +527,61 @@ func (t *msgTrace) setHopHeader(c *client, msg []byte) []byte {
 	return c.setHeader(MsgTraceHop, t.nhop, msg)
 }
 
-// Will look for the MsgTraceSendTo header and change the first character
-// to an 'X' so that if this message is sent to a remote, the remote will
-// not initialize tracing since it won't find the actual MsgTraceSendTo
-// header. The function returns the position of the header so it can
-// efficiently be re-enabled by calling enableTraceHeader.
+// Will look for the MsgTraceSendTo and trcCtx headers and change the first
+// character to an 'X' so that if this message is sent to a remote, the remote
+// will not initialize tracing since it won't find the actual trace headers.
+// The function returns the position of the headers so it can efficiently be
+// re-enabled by calling enableTraceHeaders.
 // Note that if `msg` can be either the header alone or the full message
 // (header and payload). This function will use c.pa.hdr to limit the
 // search to the header section alone.
-func (t *msgTrace) disableTraceHeader(c *client, msg []byte) int {
+func (t *msgTrace) disableTraceHeaders(c *client, msg []byte) []int {
 	// Code largely copied from getHeader(), except that we don't need the value
 	if c.pa.hdr <= 0 {
-		return -1
+		return []int{-1, -1}
 	}
 	hdr := msg[:c.pa.hdr]
-	key := stringToBytes(MsgTraceSendTo)
-	pos := bytes.Index(hdr, key)
-	if pos < 0 {
-		return -1
-	}
-	// Make sure this key does not have additional prefix.
-	if pos < 2 || hdr[pos-1] != '\n' || hdr[pos-2] != '\r' {
-		return -1
-	}
-	index := pos + len(key)
-	if index >= len(hdr) {
-		return -1
-	}
-	if hdr[index] != ':' {
-		return -1
+	headers := [2]string{MsgTraceSendTo, trcCtx}
+	positions := [2]int{}
+	for i := 0; i < 2; i++ {
+		key := stringToBytes(headers[i])
+		pos := bytes.Index(hdr, key)
+		if pos < 0 {
+			positions[i] = -1
+			continue
+		}
+		// Make sure this key does not have additional prefix.
+		if pos < 2 || hdr[pos-1] != '\n' || hdr[pos-2] != '\r' {
+			positions[i] = -1
+			continue
+		}
+		index := pos + len(key)
+		if index >= len(hdr) {
+			positions[i] = -1
+			continue
+		}
+		if hdr[index] != ':' {
+			positions[i] = -1
+			continue
+		}
+		// Disable the trace by altering the first character of the header
+		hdr[pos] = 'X'
+		positions[i] = pos
 	}
-	// Disable the trace by altering the first character of the header
-	hdr[pos] = 'X'
-	// Return the position of that character so we can re-enable it.
-	return pos
+	// Return the positions of those characters so we can re-enable the headers.
+	return positions[:2]
 }
 
 // Changes back the character at the given position `pos` in the `msg`
 // byte slice to the first character of the MsgTraceSendTo header.
-func (t *msgTrace) enableTraceHeader(c *client, msg []byte, pos int) {
-	if pos <= 0 {
-		return
+func (t *msgTrace) enableTraceHeaders(c *client, msg []byte, positions []int) {
+	headers := [2]string{MsgTraceSendTo, trcCtx}
+	for i, pos := range positions {
```

Comment:
Can inline I think.

```go
for i, pos := range [2]string{MsgTraceSendTo, trcCtx}
```

Reply:
No, we are given `positions` which are the indexes in the `msg` byte slice where to put back the first character of the 2 given headers. But agreed that I can replace `headers` with a slice of the first character of the 2 headers.